src/Repl.hs

```diff
@@ -124,23 +126,24 @@ treatSpecialInput (':':rest) =
            Nothing -> rewriteError ("Unknown special command: :" ++ [cmd])
 treatSpecialInput arg = arg
 
-repl :: String -> InputT (StateT Context IO) ()
-repl readSoFar =
+repl :: String -> String -> InputT (StateT Context IO) ()
+repl readSoFar prompt =
   do context <- get
-     let prompt = strWithColor Yellow (if null readSoFar then projectPrompt (contextProj context) else "     ")
-     input <- getInputLine prompt
+     input <- getInputLine (strWithColor Yellow prompt)
      case input of
         Nothing -> do
           liftIO exitSuccess
           return ()
         Just i -> do
           let concat = readSoFar ++ i ++ "\n"
```

Comment:
Not related to this PR, but maybe we should rename this binding? 

`concat` is a function in the Prelude, so this can throw people off if they're reading the code quickly.

Reply:
This was renamed to `concatenated`!